src/v/cluster/partition_balancer_types.h

```diff
@@ -37,23 +41,135 @@ struct node_disk_space {
     }
 };
 
-struct partition_balancer_violations {
-    struct unavailable_node {
+struct partition_balancer_violations
+  : serde::envelope<partition_balancer_violations, serde::version<0>> {
+    struct unavailable_node
+      : serde::envelope<unavailable_node, serde::version<0>> {
         model::node_id id;
-        ss::lowres_clock::duration unavailable_duration;
+        ss::lowres_clock::duration unavailable_for;
+
+        unavailable_node(model::node_id id, ss::lowres_clock::duration d)
+          : id(id)
+          , unavailable_for(d) {}
+
+        auto serde_fields() { return std::tie(id, unavailable_for); }
     };
 
-    struct full_node {
+    struct full_node : serde::envelope<full_node, serde::version<0>> {
         model::node_id id;
         uint32_t disk_used_percent;
+
+        full_node(model::node_id id, uint32_t disk_used_percent)
+          : id(id)
+          , disk_used_percent(disk_used_percent) {}
+
+        auto serde_fields() { return std::tie(id, disk_used_percent); }
     };
 
     std::vector<unavailable_node> unavailable_nodes;
     std::vector<full_node> full_nodes;
 
+    partition_balancer_violations() noexcept = default;
+
+    partition_balancer_violations(
+      std::vector<unavailable_node> un, std::vector<full_node> fn)
+      : unavailable_nodes(std::move(un))
+      , full_nodes(std::move(fn)) {}
+
+    auto serde_fields() { return std::tie(unavailable_nodes, full_nodes); }
+
     bool is_empty() const {
         return unavailable_nodes.empty() && full_nodes.empty();
     }
 };
 
+struct partition_balancer_overview_request
+  : serde::envelope<partition_balancer_overview_request, serde::version<0>> {
+    auto serde_fields() { return std::tie(); }
+};
+
+struct partition_balancer_overview_reply
+  : serde::envelope<partition_balancer_overview_reply, serde::version<0>> {
+    errc error;
+
+    std::optional<partition_balancer_violations> violations;
+    model::timestamp last_tick_time;
+
+    auto serde_fields() { return std::tie(error, violations, last_tick_time); }
+};
+
 } // namespace cluster
+
+namespace reflection {
```

Comment:
I think recent guidance is that we don't need to define adls for new messages that start out with serde support, cc @dotnwat

Reply:
IIUC this depends on https://github.com/redpanda-data/redpanda/pull/4963 which is still in unmerged state. I'd be happy to drop adl support :)

Reply:
talked with Noah, that PR might land on Wednesday the earliest, so propose to add serde support in a separate PR and get this merged earlier